### ISSUE TITLE
Bike/Bus/Parking Lane restrictions

### ISSUE THREAD

**Issue by JukEboXAuDiO** [CitiesSkylinesMods/TMPE#1473]
### Describe your idea
With the current TMPE version you can select which lane is restricted to what. Would it be possible to restrict a lane to just bikes? Also allow for lanes for parking. Inline or not. Sidewalks are relegated to pedestrians. This way you can make roads for cars/buses/parking/buses. 

This allows you to make inner city lanes more customizable with larger blank roads. Then you can customize blank roads so much better.

### Screenshots?
So you can customize roads something like this. https://steamcommunity.com/sharedfiles/filedetails/?id=1226055866&searchtext=bike+bus+roads

**Comment by krzychu124** [comment]
> With the current TMPE version you can select which lane is restricted to what.

True, but we are only restricting lane usage. It's currently not possible to add vehicle type to lane without cloning/creating new asset. When you have option to ban specific vehicles it means that lane has vehicle type `Car` which is covering _service vehicles, busses, cargo trucks and passenger vehicles_. 

Bikes, trams, trains are separate vehicle types that needs to be explicitly set in the asset configuration.
Similar for parking lanes. 

Lane is explicitly set to be type of `Parking` instead of `Vehicle` 
![image](https://user-images.githubusercontent.com/19638970/159128149-bba0cbbe-3177-492c-85a1-4a760392f65e.png)

`vehicleType` describes which vehicles are allowed to use that lane(see below)

![image](https://user-images.githubusercontent.com/19638970/159128194-9e44d029-eccd-4e38-b44d-46cdbcb9a907.png)

So when it's set `Parking` + `Car`, then all vehicles from group `Car` can park but cannot drive on it, and when `Vehicle` + `Car` then vehicles from group `Car` can use it to drive, but cannot park, obviously.

**Comment by aubergine10** [comment]
Can lanes be turned on/off?

For example, in loading phase, add extra lanes to road prefabs, on top of existing lanes, so that each "stack" of lanes contains full set of variants (parking vs. vehicle, different directions, etc). By default the added lanes are disabled, but mod UI can allow user to cycle through which lane is active for each stack, per segment (somehow).

* All existing roads, including vanilla, can be extended by this approach (if feasible, ofc)
* If mod is removed, roads return to normal functionality
